src/hip/HIPStream.cpp

```diff
@@ -186,32 +198,32 @@ template <typename T>
 __global__ void nstream_kernel(T * a, const T * b, const T * c)
 {
   const T scalar = startScalar;
-  const int i = hipBlockDim_x * hipBlockIdx_x + hipThreadIdx_x;
+  const size_t i = threadIdx.x + blockIdx.x * blockDim.x;
   a[i] += b[i] + scalar * c[i];
 }
 
 template <class T>
 void HIPStream<T>::nstream()
 {
-  hipLaunchKernelGGL(HIP_KERNEL_NAME(nstream_kernel<T>), dim3(array_size/TBSIZE), dim3(TBSIZE), 0, 0, d_a, d_b, d_c);
+  nstream_kernel<T><<<dim3(array_size/TBSIZE), dim3(TBSIZE), 0, 0>>>(d_a, d_b, d_c);
   check_error();
   hipDeviceSynchronize();
   check_error();
 }
 
-template <class T>
+template <typename T>
 __global__ void dot_kernel(const T * a, const T * b, T * sum, int array_size)
 {
   __shared__ T tb_sum[TBSIZE];
 
-  int i = hipBlockDim_x * hipBlockIdx_x + hipThreadIdx_x;
-  const size_t local_i = hipThreadIdx_x;
+  const size_t local_i = threadIdx.x;
+  size_t i = blockDim.x * blockIdx.x + local_i;
 
   tb_sum[local_i] = 0.0;
-  for (; i < array_size; i += hipBlockDim_x*hipGridDim_x)
+  for (; i < array_size; i += blockDim.x*gridDim.x)
```

Comment:
Indeed - this loop here should mean that the kernel is valid even if the block size does not evenly divide...